### DISCUSSION TITLE
Relay to protection.outlook.com ends in: TenantAttribution; Relay Access Denied [ValidationStatus of '' is EmptyCertificate]

### DISCUSSION THREAD

**Discussion by cvillabrille** [postalserver/postal#3261] [Help with using Postal]
Due to subnet division we are trying to configure an email server using _postal_ for several internal company domains.

Postal will be the only one having external access to Outlook (servers does not have internet connection), using, as relay for any external address, an Outlook server (protection.outlook.com).

When we setup the smtp_relays using starttls at Postal, we receive the error from outlook like:

`TenantAttribution; Relay Access Denied [ValidationStatus of '' is EmptyCertificate]
` 

We have tried several different setups in term of: no TLS, TLS only or STARTTLS  as follows, reporting the following errors:

- Without specifying parameters on smtp_relay:

\#  smtp_relays: smtp://xxx.mail.protection.outlook.com:25
\#  smtp_relays: smtp://xxx.mail.protection.outlook.com:25?ssl_mode=tls

We see the error:
`Net::SMTPFatalError: 550 5.7.64 TenantAttribution; Relay Access Denied [No TLS] [xxx.prod.outlook.com]
`

- Specifying as below examples
\#  smtp_relays: smtp://xxx.mail.protection.outlook.com:25?ssl_mode=Auto
\#  smtp_relays: smtp://xxx.mail.protection.outlook.com:25?ssl_mode=starttls
\#  smtp_relays: smtp://xxx.mail.protection.outlook.com:25?starttls=when-available
\#  smtp_relays: smtp://xxx.mail.protection.outlook.com:587?ssl_mode=Auto
\#  smtp_relays: smtp://xxx.mail.protection.outlook.com:25?starttls=Auto
\#  smtp_relays: smtp://xxx.mail.protection.outlook.com:25?tls=Auto

we see:
`Net::SMTPFatalError: 550 5.7.64 TenantAttribution; Relay Access Denied [ValidationStatus of '' is EmptyCertificate] [xxx.prod.outlook.com]`

Hence the question is:

- How we can setup smtp_relays correctly using STARTTLS and not receiving the above mentioned error? (is a matter of modifiers of smtp_relay or it's something else).

Documentation is not very clear about the smtp_relays parameter *(sorry about that). 

I found as reference:

Microsoft definition of the error:
https://learn.microsoft.com/en-us/exchange/troubleshoot/email-delivery/ndr/relay-access-denied-smtp
(Postal's IP is included in the Allowed IPs).

STARTTLS setup:
https://github.com/orgs/postalserver/discussions/2769

https://github.com/postalserver/postal/issues/48

https://github.com/orgs/postalserver/discussions/2912

https://github.com/orgs/postalserver/discussions/2886

https://github.com/postalserver/postal/issues/277

Conf file V3 parameters:
https://github.com/postalserver/postal/blob/main/doc/config/yaml.yml

**Comment by cvillabrille** [comment]
Quick update.

If I set the  DNS with:

dns:
  return_path_domain: example.com

I can forward emails to outside using the email relay configured with (No error  of _TenantAttribution_ ).. 
  smtp_relays: smtp://xxxxmail.protection.outlook.com:25?ssl_mode=Auto


However, then I cannot send emails to any of the domains configured in the postal server. I always receive the error:

smtp: Failed: 550 Invalid server token


How can I configure postal to allow forwarding emails to the smtp_relays and also to the domains configured inside Postal?

Much appreciated in advance. 

PS:
I have tested stopping postal, enabling postfix or nullmailer in the host without any setup, I can forward emails to the smtp relay server.

**Comment by willpower232** [comment]
Are you specifying these outlook.com's as relays in the config file? Those ones are mostly for when your Postal is in some kind of private network and there is an email gateway to the outside internet.

If you're trying to ensure that Postal can only email specific email addresses, that isn't really a feature of Postal and should be set up in whatever software you're using to create the emails.

If not, can you tell us a bit more about what you are trying to do?

**Reply by cvillabrille** [reply]
Apologizes for the delay, at X-mas time is always hard to be online
> Are you specifying these outlook.com's as relays in the config file? Those ones are mostly for when your Postal is in some kind of private network and there is an email gateway to the outside internet.

it is configured in the postal: section with 
```
postal:
   smtp_relays: smtp://xxxx.mail.protection.outlook.com:25?ssl_mode=Auto
```


> If not, can you tell us a bit more about what you are trying to do?
The idea is that Postal will forward to other email servers configured in the routing section, and the rest to be forwarded to the email_relay.


After further investigations, I realized all is related to the return_path_domain. 

Configuring another alternative domain in the return_path_domain, outlook allows to forward emails coming from Postal (since the official email server is outlook for domain.com).

```
dns:
  mx_records:
    - mailrelay.domain.com
  spf_include: spf.mail.domain.com
  return_path_domain: something.else.resolvable.by.dns.and.allowed.in.exchange.setup
  route_domain: routes.mailrelay.domain.com
  track_domain: track.mailrelay.domain.com
```

**Reply by willpower232** [reply]
The smtp_relays section is for all outgoing messages from Postal, I don't think microsoft will let you relay through them too much :sweat_smile:

**Comment by cvillabrille** [comment]
I agree :) but we are “relaying” to official domain emails and Postal is one of the allowed forwarders at Microsoft side.

It was a tricky issue considering the culprit was related to the DNS setup IN the application. While normally is at the official DNS side.

The fact that updating that entry made it work was hard to figure it out. And the error was also not very descriptive. Documentation improvement?
